src/cc/frontends/clang/b_frontend_action.cc

```diff
@@ -487,6 +488,52 @@ bool ProbeVisitor::VisitMemberExpr(MemberExpr *E) {
   rewriter_.ReplaceText(expansionRange(SourceRange(member, E->getLocEnd())), post);
   return true;
 }
+bool ProbeVisitor::VisitArraySubscriptExpr(ArraySubscriptExpr *E) {
+  if (memb_visited_.find(E) != memb_visited_.end()) return true;
+  if (!ProbeChecker(E, ptregs_, track_helpers_).needs_probe()) {
+    return true;
+  }
+
+  // parent expr has addrof, skip the rewrite, set is_addrof_ to flase so
+  // it won't affect next level of indirect address
+  if (is_addrof_) {
+    return true;
+  }
+
+  Expr *base = E->getBase();
```

Comment:
The base or better E may not be rewritable, we need need a check
```
  if (!rewriter_.isRewritable(E->getLocStart()))
     return true;
```

Reply:
Done.